src/NexusMods.DataModel/Loadouts/LoadoutRegistry.cs

```diff
@@ -456,4 +458,65 @@ public bool Contains(LoadoutId loadoutId)
     {
         return GetId(loadoutId) != null;
     }
+
+    /// <inheritdoc />
+    public void Delete(LoadoutId loadoutId)
+    {
+        if (_isDisposed)
+            throw new ObjectDisposedException(nameof(LoadoutRegistry));
+
+        // Get all revisions 
+        var loadout = Get(loadoutId);
+        if (loadout == null)
+            throw new ArgumentException("Loadout not found", nameof(loadoutId));
+
+        // Remove LoadoutId
+        var dbId = loadoutId.ToEntityId(EntityCategory.LoadoutRoots);
+        _store.Delete(dbId);
+        _logger.LogInformation("Loadout {LoadoutId} deleted", loadoutId);
+
+        // Remove all previous revisions.
+        var currentVersion = loadout.PreviousVersion;
+        while (true)
+        {
+            var previousLoadout = currentVersion.Value;
+            if (previousLoadout == null)
+                break;
+
+            _store.Delete(previousLoadout.DataStoreId);
+            currentVersion = previousLoadout.PreviousVersion;
+        }
+        
+        _markers.TryRemove(loadoutId, out _);
+        _loadoutsIds.Remove(loadoutId);
+        _cache.Edit(x => x.Remove(loadoutId));
+    }
+
+    /// <inheritdoc />
+    public bool IsActive(GameInstallation installation, LoadoutId id, IId? lastAppliedId)
+    {
+        var loadout = Get(id);
+        if (loadout == null)
+            throw new ArgumentException("Loadout not found", nameof(id));
+
+        // Check if any of the loadout versions is the currently active loadout
+        var isActiveLoadout = loadout.DataStoreId.Equals(lastAppliedId);
+        var currentVersion = loadout.PreviousVersion;
+        while (!currentVersion.Id.Equals(IdEmpty.Empty))
+        {
+            if (currentVersion.Id.Equals(lastAppliedId))
+            {
+                isActiveLoadout = true;
+                break;
+            }
+
+            var value = currentVersion.Value;
+            if (value == null)
+                break;
+
+            currentVersion = value.PreviousVersion;
+        }
+
+        return isActiveLoadout;
```

Comment:
Shouldn't this be just two lines?

```csharp
var loadout = GetLoadout(lastAppliedId);
return loadout.LoadoutId = id;
```

Reply:
I renamed this to `IsApplied` in retrospect, because the original `IsActive` may be misleading depending on context.

In any case, the situation here is that the last applied loadout is not always the latest revision of a loadout.
For instance, suppose you make a change to the currently applied loadout.

The latest revision is updated, but until you actually apply, the last applied ID is of a previous revision. This is why we need to check all revisions just in case.

Reply:
> last applied loadout is not always the latest revision of a loadout

But we don't care about that, at least I don't think we do. By definition, the last **applied** loadout was applied at some point. It's not a method that checks if the last applied loadout was applied, that's kinda weird. It's a method that checks if the last applied revision was part of the given loadout. So it should be just checking whether the loadout IDs match. No need to walk all revisions.

Reply:
erri is right, from a loadout revision Id you can get the LoadoutId with GetLoadout(revisionId).LoadoutId. 
Then you just compare the LoadoutId with the passed one. No need to check rev by rev

Reply:
Yeah pretty much.

The problem here is a bit interesting.
Basically, we don't store the actual `LoadoutId` (unique ID for loadout) of the last applied loadout anywhere. Not in Disk State, not as Last Applied Loadout. etc. Instead we store the `LoadoutRevision`, which is the unique ID for revision of a loadout. 

And there's no mapping anywhere of `LoadoutRevision` to `LoadoutId` that I'm aware of.

Should I maybe rename it to `IsAnyRevisionApplied`?
Might be more descriptive.

Reply:
The mapping is there and is this: you have a `RevisionId`
`LoadoutRegistry.GetLoadout(RevisionId)` -> `Loadout`
`Loadout.LoadoutId` - > `LoadoutId`

Reply:
Yes our names are a mess: https://github.com/Nexus-Mods/NexusMods.App/issues/1204

Reply:
I posted at the same time as you :sweat_smile: .
That's why I edited my post with `wait a moment`.

Seems you're right. I was always under the impression that a revision was a separate item, and not just an ID to a previous loadout; hence the confusion.

Let me patch this up.

Reply:
No worries, I'm not surprised you were confused, it really isn't intuitive.